libvast/vast/system/index.hpp

```diff
@@ -256,19 +273,20 @@ caf::expected<flatbuffers::Offset<fbs::Index>>
 pack(flatbuffers::FlatBufferBuilder& builder, const index_state& state);
 
 /// Indexes events in horizontal partitions.
-/// @param store The global store actor. Not used by the index itself but
-/// forwarded to partitions.
 /// @param filesystem The filesystem actor. Not used by the index itself but
 /// forwarded to partitions.
 /// @param dir The directory of the index.
+/// @param partition_local_stores Whether to use a global store for segments.
 /// @param partition_capacity The maximum number of events per partition.
 /// @param taste_partitions How many lookup partitions to schedule immediately.
 /// @param num_workers The maximum amount of concurrent lookups.
 /// @param meta_index_fp_rate The false positive rate for the meta index.
 /// @pre `partition_capacity > 0
+//  TODO: Use a settings struct for the various parameters.
 index_actor::behavior_type
-index(index_actor::stateful_pointer<index_state> self, store_actor store,
-      filesystem_actor filesystem, const std::filesystem::path& dir,
+index(index_actor::stateful_pointer<index_state> self,
+      filesystem_actor filesystem, archive_actor archive,
```

Comment:
Prefer to cast in `spawn_index()`:
```suggestion
      filesystem_actor filesystem, store_builder_actor global_store,
```
Can you also align the comments?

Reply:
Not sure about this one, we certainly want the one and only archive here, not just any store builder.

Reply:
That would be OK as well, but use the `archive` type in the `index_state` as well then.